utils/build_swift/tests/expected_options.py

```diff
@@ -482,6 +483,7 @@ class IgnoreOption(_BaseOption):
     AppendOption('--extra-swift-args', dest='extra_swift_args'),
     AppendOption('--stdlib-deployment-targets',
                  dest='stdlib_deployment_targets'),
+    AppendOption('--test-paths', dest='test_paths'),
```

Comment:
Thanks for adding the test option here, have you tried running the test suite from the `utils` directory with the command `python unittest discover -m build_swift/tests`? (Ignore the failing preset test, I already know about it. 😅)

Reply:
This is what I've got after running the tests:

```
======================================================================
FAIL: test_default_value_build_jobs (test_driver_arguments.buildbot,tools=RA,stdlib=RD,single-thread)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/xedin/work/swift/swift/utils/build_swift/tests/test_driver_arguments.py", line 126, in test
    .format(dest, default_value, parsed_value))
AssertionError: Invalid default value for "build_jobs": 4 != 8

======================================================================
FAIL: test_default_value_llvm_max_parallel_lto_link_jobs (test_driver_arguments.buildbot,tools=RA,stdlib=RD,single-thread)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/xedin/work/swift/swift/utils/build_swift/tests/test_driver_arguments.py", line 126, in test
    .format(dest, default_value, parsed_value))
AssertionError: Invalid default value for "llvm_max_parallel_lto_link_jobs": 0 != 2

======================================================================
FAIL: test_default_value_swift_tools_max_parallel_lto_link_jobs (test_driver_arguments.buildbot,tools=RA,stdlib=RD,single-thread)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/xedin/work/swift/swift/utils/build_swift/tests/test_driver_arguments.py", line 126, in test
    .format(dest, default_value, parsed_value))
AssertionError: Invalid default value for "swift_tools_max_parallel_lto_link_jobs": 0 != 1

======================================================================
FAIL: test_preset_buildbot_iphoneos_arm64_crosscompiler (test_driver_arguments.buildbot,tools=RA,stdlib=RD,single-thread)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/xedin/work/swift/swift/utils/build_swift/tests/test_driver_arguments.py", line 281, in test
    preset_name, e))
AssertionError: Failed to parse preset "buildbot_iphoneos_arm64_crosscompiler": ("failed to parse arguments: ['--build-subdir=LLVMClangSwift_iphoneos', '--release', '--', '--cross-compile-hosts=iphoneos-arm64', '--compiler-vendor=apple', '--swift-primary-variant-sdk=IOS', '--swift-primary-variant-arch=arm64', '--extra-cmake-options=-DLLVM_ENABLE_BACKTRACES:BOOL=OFF', '--build-toolchain-only=1', '--build-swift-static-stdlib=0', '--build-swift-dynamic-stdlib=0', '--build-swift-static-sdk-overlay=0', '--build-swift-dynamic-sdk-overlay=0', '--build-runtime-with-host-compiler=0', '--skip-build-benchmarks', '--llvm-include-tests=0', '--llvm-enable-modules=0', '--build-swift-ios-test=0', '--build-swift-examples=0', '--swift-include-tests=0', '--darwin-deployment-version-ios=10.0', '--darwin-crash-reporter-client=1']", ValueError('error: unknown setting: build-swift-ios-test',))

----------------------------------------------------------------------
Ran 404 tests in 118.604s

FAILED (failures=4)
```

Does that look ok?

Reply:
The last test is the one I expected to fail, the first 3 look like completely separate issues with the default values I gathered for the initial suite. Most likely those values are determined for each machine and my development machine defaulted to different values than yours.